src/pages/index.js

```diff
@@ -5,27 +5,39 @@ import DefaultLayout from "../layouts/default"
 import Image from "../components/image"
 import SEO from "../components/seo"
 
+import { groupBy, getDateYear } from "../utils"
+
 const IndexPage = ({ data }) => {
   // all posts without dates are assumed to be drafts or pages
   // not to be added to postsList
   const posts = data.allMarkdownRemark.edges.filter(
     p => p.node.frontmatter.date !== null
   )
-  const postsList = posts.map(post => (
-    <li key={post.node.id}>
-      <div className="post-date code">
-        <small>{post.node.frontmatter.date}</small>
-      </div>
-      <div className="title">
-        <Link to={post.node.fields.slug}>{post.node.frontmatter.title}</Link>
+  const postsList = posts =>
+    posts.map(post => (
+      <li key={post.node.id}>
+        <div className="post-date code">
+          <small>{post.node.frontmatter.date}</small>
+        </div>
+        <div className="title">
+          <Link to={post.node.fields.slug}>{post.node.frontmatter.title}</Link>
+        </div>
+      </li>
+    ))
+
+  const postsListContainer = groupBy(posts, getDateYear).map(
+    ({ year, posts }, i) => (
+      <div key={i}>
+        <b>{year}</b>
```

Comment:
All it needs is some styling and we'll be good too 🚀. Let me work on it. Thanks!

Reply:
Awesome 👍